lib/ts/recipe/accountlinking/recipe.ts

```diff
@@ -131,184 +136,129 @@ export default class Recipe extends RecipeModule {
     }
 
     // this function returns the user ID for which the session will be created.
-    createPrimaryUserIdOrLinkAccounts = async ({
+    createPrimaryUserIdOrLinkByAccountInfoOrLinkToSessionIfProvided = async ({
         tenantId,
-        user,
+        inputUser,
+        recipeUserId,
+        session,
         userContext,
     }: {
         tenantId: string;
-        user: User;
+        inputUser: User;
+        recipeUserId: RecipeUserId;
+        session: SessionContainerInterface | undefined;
         userContext: UserContext;
-    }): Promise<User> => {
+    }): Promise<
+        | { status: "OK"; user: User }
+        | {
+              status: "LINKING_TO_SESSION_USER_FAILED";
+              reason:
+                  | "EMAIL_VERIFICATION_REQUIRED"
+                  | "RECIPE_USER_ID_ALREADY_LINKED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR"
+                  | "ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR";
+          }
+        | {
+              status: "NON_PRIMARY_SESSION_USER";
+              reason: "ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR";
```

Comment:
we could call it something like `NON_PRIMARY_SESSION_USER_WHEN_LINKING`, but this could also come from preAuth checks, where we haven't attempted to link yet. This can happen if the user called with a session, shouldDoAutomaticAccounLinking returns true + the verification status checks out but there is a conflicting primary user.